### ISSUE TITLE
Example use with Typescript

### ISSUE THREAD

**Issue by stoivo** [EMH333/esbuild-svelte#133]
Hi, Thanks for all your work on this project. 

**Describe the bug**

I wanted to setup esbuild with svelt and typescript I tried to follow you guidelines as best as possible. I have some issue to import typescript files into a svelt component. I forked the project and added an example-ts. https://github.com/stoivo/esbuild-svelte

Esbuild seams to complain abount Joke and svelt-check seams unhappy too. 

<details><summary>Svelt-check</summary>

```
⋊> ~/D/e/example-ts on main ⨯ pnpm svelte-check

====================================
Loading svelte-check in workspace: /Users/simon/Downloads/esbuild-svelte/example-ts
Getting Svelte diagnostics...

/Users/simon/Downloads/esbuild-svelte/example-ts/index.svelte:5:3
Error: This import is never used as a value and must use 'import type' because 'importsNotUsedAsValues' is set to 'error'. (ts)
  import {haha} from "$fun";
  import {Joke} from "$fun";
  import Second from "./second.svelte";


/Users/simon/Downloads/esbuild-svelte/example-ts/index.svelte:5:11
Error: 'Joke' is a type and must be imported using a type-only import when 'preserveValueImports' and 'isolatedModules' are both enabled. (ts)
  import {haha} from "$fun";
  import {Joke} from "$fun";
  import Second from "./second.svelte";


/Users/simon/Downloads/esbuild-svelte/example-ts/index.svelte:7:7
Error: Variable 'inputBinding' implicitly has type 'any' in some locations where its type cannot be determined. (ts)
  import Second from "./second.svelte";
  let inputBinding;



/Users/simon/Downloads/esbuild-svelte/example-ts/index.svelte:18:5
Error: Variable 'inputBinding' implicitly has an 'any' type. (ts)
  function change() {
    inputBinding.value = "testing" + Math.round(Math.random() * 100);
  }


====================================
svelte-check found 4 errors, 0 warnings, and 0 hints



```
</details>

<details><summary>esbuild</summary>

```
⋊> ~/D/e/example-ts on main ⨯ pnpm build

> @ build /Users/simon/Downloads/esbuild-svelte/example-ts
> node buildscript.js

 > index.svelte:25:9: error: No matching export in "fun.ts" for import "Joke"
    25 │ import { Joke } from "$fun";
       ╵          ~~~~

1 error
Errors:  Error: Build failed with 1 error:
index.svelte:25:9: error: No matching export in "fun.ts" for import "Joke"
    at failureErrorWithLog (/Users/simon/Downloads/esbuild-svelte/example-ts/node_modules/.pnpm/esbuild@0.12.29/node_modules/esbuild/lib/main.js:1475:15)
    at /Users/simon/Downloads/esbuild-svelte/example-ts/node_modules/.pnpm/esbuild@0.12.29/node_modules/esbuild/lib/main.js:1133:28
    at runOnEndCallbacks (/Users/simon/Downloads/esbuild-svelte/example-ts/node_modules/.pnpm/esbuild@0.12.29/node_modules/esbuild/lib/main.js:923:63)
    at buildResponseToResult (/Users/simon/Downloads/esbuild-svelte/example-ts/node_modules/.pnpm/esbuild@0.12.29/node_modules/esbuild/lib/main.js:1131:7)
    at /Users/simon/Downloads/esbuild-svelte/example-ts/node_modules/.pnpm/esbuild@0.12.29/node_modules/esbuild/lib/main.js:1240:14
    at /Users/simon/Downloads/esbuild-svelte/example-ts/node_modules/.pnpm/esbuild@0.12.29/node_modules/esbuild/lib/main.js:611:9
    at handleIncomingPacket (/Users/simon/Downloads/esbuild-svelte/example-ts/node_modules/.pnpm/esbuild@0.12.29/node_modules/esbuild/lib/main.js:708:9)
    at Socket.readFromStdout (/Users/simon/Downloads/esbuild-svelte/example-ts/node_modules/.pnpm/esbuild@0.12.29/node_modules/esbuild/lib/main.js:578:7)
    at Socket.emit (node:events:390:28)
    at addChunk (node:internal/streams/readable:315:12) {
  errors: [
    {
      detail: undefined,
      location: [Object],
      notes: [],
      pluginName: '',
      text: 'No matching export in "fun.ts" for import "Joke"'
    }
  ],
  warnings: []
} undefined
```
</details>

**To Reproduce**
1. clone https://github.com/stoivo/esbuild-svelte
1. cd example-ts
1. pnpm install or npm install
1. pnpm build
1. pnpm typecheck

**Comment by stoivo** [comment]
After a couple more hours of fiddeling I think I got it. I would very much like to have it confirmed by by and I think we could save the next person some time if we document it a bit. 

It seam like I have to do type import separate from value import like 

```js
  import {haha} from "$fun";
  import type {Joke} from "$fun";
```
It would be good to find this in the official docs. I  tested with and sveltkitinstaller and I had to split the import over two lines. I used `pnpm create testofficialsv` to create the project.

Even if this is working do you know if I can do relative paths like "./fun" instead of "$fun"

**Comment by EMH333** [comment]
Yep that was the solution I was going to recommend. I'm not totally sure how well documenting this integrates with esbuild-svelte (the issue you describe is technically svelte-preprocess related). If you have a succinct and elegant way to document it in this project, feel free to open a PR!

With regards to `./fun` vs `$fun`, I'll admit I've never seen the dollar sign syntax before

**Comment by stoivo** [comment]
I make it work without $fun

**Comment by EMH333** [comment]
This has been resolved with #134
